src/api/app/views/webui/request/_changes_add_role.html.haml

```diff
@@ -0,0 +1,15 @@
+:ruby
+  parameters = { project: action.target_project, request_action_id: action.id, notification_id: current_notification }
+  roles_link = if action.target_package
+                  package_users_path(parameters.merge!(package: action.target_package))
+                else
+                  project_users_path(parameters)
+                end
```

Comment:
Could we move this into an per-action helper?

Reply:
Yeah, the `parameters` are not used in the partial, right? Why don't you build the parameters inside the `roles_link` function also?